contrib/dump-mtd-ifd.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2024 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+#
+# pylint: disable=invalid-name,missing-docstring,too-few-public-methods,too-many-locals
+
+import sys
+import struct
+import io
+from typing import List, Optional
+import argparse
+
+
+class IfdPartition:
+    REGION_NAMES = [
+        "desc",
+        "bios",
+        "me",
+        "gbe",
+        "platform",
+        "devexp",
+        "bios2",
+        "ec",
+        "ie",
+        "10gbe",
+    ]
+
+    def __init__(self, region: int = 0, offset: int = 0, size: int = 0) -> None:
+        self.region: int = region
+        self.offset: int = offset
+        self.size: int = size
+
+    def __str__(self) -> str:
+        return (
+            "IfdPartition("
+            f"region={IfdPartition.REGION_NAMES[self.region]}, "
```

Comment:
you probably want a safety check here to check if the region is something that's not in `REGION_NAMES`

Reply:
Good idea, added.